### PR TITLE
vxlan_global --> overlay_global + fixes

### PR DESCRIPTION
**Pull request by glennmatthews**
- Rename vxlan_global to overlay_global per discussion.
- As some properties are dependent on 'nv overlay evpn' and others are not, move the feature logic out of the initialize() method and into the appropriate getters and setters
- More thorough testing

I need to validate this against a I2 image still but here it is against an I3 image:

```
Node under test:
  - name  - n9k-108
  - type  - N9K-C9396PX
  - image - bootflash:///nxos.7.0.3.I3.0.278.bin

TestOverlayGlobal#test_dup_host_ip_addr_detection = 5.25 s = .
TestOverlayGlobal#test_dup_host_mac_detection = 3.02 s = .
TestOverlayGlobal#test_anycast_gateway_mac = 3.99 s = .

Fabulous run in 12.253600s, 0.2448 runs/s, 1.6322 assertions/s.

3 runs, 20 assertions, 0 failures, 0 errors, 0 skips
```

### CODE REVIEW

**Comment by glennmatthews** [comment]
```
Node under test:
  - name  - n9k-108
  - type  - N9K-C9396PX
  - image - bootflash:///nxos.7.0.3.I2.1.bin

TestOverlayGlobal#test_anycast_gateway_mac = 9.02 s = .
TestOverlayGlobal#test_dup_host_mac_detection = 12.93 s = .
TestOverlayGlobal#test_dup_host_ip_addr_detection = 3.54 s = .

Fabulous run in 25.486725s, 0.1177 runs/s, 0.7847 assertions/s.

3 runs, 20 assertions, 0 failures, 0 errors, 0 skips
```

**Comment by glennmatthews** [comment]
Test passes against both I2 and I3 on N9k.

File: lib/cisco_node_utils/feature.rb [hunk 1]
```diff
@@ -85,11 +85,10 @@ def self.nv_overlay_evpn_enable
       # both features; therefore feature fabric-forwarding is best-effort
       # and ignored on cli failure.
       begin
-        return if nv_overlay_evpn_enabled?
         config_set('feature', 'fabric_forwarding')
-        config_set('feature', 'nv_overlay_evpn')
+      rescue Cisco::CliError
+        CiscoLogger.debug 'feature fabric-forwarding CLI was rejected'
```
Comment (mikewiebe) [inline lib/cisco_node_utils/feature.rb]: _nit_ It's actually `feature fabric forwarding`, no `-`

File: lib/cisco_node_utils/feature.rb [hunk 2]
```diff
@@ -85,11 +85,10 @@ def self.nv_overlay_evpn_enable
       # both features; therefore feature fabric-forwarding is best-effort
       # and ignored on cli failure.
       begin
-        return if nv_overlay_evpn_enabled?
         config_set('feature', 'fabric_forwarding')
-        config_set('feature', 'nv_overlay_evpn')
+      rescue Cisco::CliError
```
Comment (mikewiebe) [inline lib/cisco_node_utils/feature.rb:89]: Can you add a comment to indicate why we are rescuing this error?

Update: Never mind... the comment above is sufficient.

**Comment by mikewiebe** [comment]
@glennmatthews Minor nit :+1:

File: lib/cisco_node_utils/feature.rb (hunk 1 above)
Comment (glennmatthews) [inline lib/cisco_node_utils/feature.rb]: Fixed.

**Comment by glennmatthews** [comment]
n3k I3:

```
Node under test:
  - name  - n3k-105
  - type  - N3K-C3048TP-1GE
  - image - bootflash:///nxos.7.0.3.I3.0.285.bin

TestOverlayGlobal#test_anycast_gateway_mac = 3.11 s = S
TestOverlayGlobal#test_dup_host_ip_addr_detection = 1.06 s = S
TestOverlayGlobal#test_dup_host_mac_detection = 2.55 s = .

Fabulous run in 6.721668s, 0.4463 runs/s, 1.6365 assertions/s.

  1) Skipped:
TestOverlayGlobal#test_anycast_gateway_mac [/home/glmatthe/cisco-network-node-utils/tests/basetest.rb:103]:
Feature 'feature', attribute 'fabric_forwarding', operation 'config_set' is unsupported on this node


  2) Skipped:
TestOverlayGlobal#test_dup_host_ip_addr_detection [/home/glmatthe/cisco-network-node-utils/tests/basetest.rb:103]:
Feature 'feature', attribute 'fabric_forwarding', operation 'config_set' is unsupported on this node

3 runs, 11 assertions, 0 failures, 0 errors, 2 skips
```

**Pull request merged**